packages/orchestrator-ui-components/src/components/WFOSteps/WFOStepList/WFOStepList.tsx

```diff
@@ -1,130 +1,291 @@
+import { Step, StepStatus } from '../../../types';
+import { WFOStep } from '../WFOStep';
 import { getStyles } from '../getStyles';
 import { useOrchestratorTheme } from '../../../hooks';
+import { stateDelta } from '../../../utils';
+import { WFOCode, WFOEyeFill } from '../../../icons';
+
+export type WFOStepListRef = {
+    scrollToStep: (stepId: string) => void;
+};
 
 export interface WFOStepListProps {
     steps: Step[];
     startedAt: string;
 }
 
-export const WFOStepList = ({ steps = [], startedAt }: WFOStepListProps) => {
-    const { theme } = useOrchestratorTheme();
-    const t = useTranslations('processes.steps');
-    let stepStartTime = startedAt;
-
-    const allDetailsClosedState = new Map(
-        steps.map((_, index) => [index, false]),
-    );
-    const allDetailsOpenedState = new Map(
-        steps.map((_, index) => [index, true]),
-    );
-
-    const [stepDetailStates, setStepDetailStates] = useState<
-        Map<number, boolean>
-    >(new Map(allDetailsClosedState));
-
-    const openAllDetails = () =>
-        setStepDetailStates(new Map(allDetailsOpenedState));
-
-    const toggleStepDetailIsOpen = (index: number) => {
-        if (stepDetailStates.has(index)) {
-            setStepDetailStates(
-                new Map(
-                    stepDetailStates.set(index, !stepDetailStates.get(index)),
-                ),
-            );
-        }
-    };
-
-    const {
-        stepSpacerStyle,
-        stepListHeaderStyle,
-        stepListContentStyle,
-        stepListContentBoldTextStyle,
-        stepListContentAnchorStyle,
-    } = getStyles(theme);
-
-    return (
-        <>
-            <EuiFlexGroup css={stepListHeaderStyle}>
-                <EuiFlexGroup css={stepListContentStyle}>
-                    <EuiText css={stepListContentBoldTextStyle}>
-                        {t('steps')}
-                    </EuiText>
-                    <EuiText
-                        css={stepListContentAnchorStyle}
-                        onClick={openAllDetails}
-                    >
-                        {t('expandAll')}
-                    </EuiText>
-                </EuiFlexGroup>
-                <EuiFlexGroup
-                    justifyContent="flexEnd"
-                    direction="row"
-                    css={{ flexGrow: 0 }}
-                    gutterSize="s"
-                >
-                    <EuiButton
-                        onClick={(
-                            e: React.MouseEvent<
-                                HTMLButtonElement | HTMLElement,
-                                MouseEvent
-                            >,
-                        ) => {
-                            e.preventDefault();
-                            alert('TODO: Implement Show delta');
-                        }}
-                        iconSide="right"
-                        size="s"
-                        iconType="visVega"
+export const WFOStepList = React.forwardRef(
```

Comment:
Break component up in smaller pieces and try to apply declarative way of coding as much as possible

Reply:
WIP